src/Input/index.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+
+const getHTMLTagForType = (type) => {
+  if (type === 'select' || type === 'textarea') return type;
+  return 'input';
+};
+
+const getClassNameForType = (inputType) => {
+  if (inputType === 'file') return 'form-control-file';
+  if (inputType === 'checkbox') return 'form-check-input';
+  if (inputType === 'radio') return 'form-check-input';
+  return 'form-control';
+};
+
+const renderOptions = options => options.map(({
+  value, label, group, ...attributes
+}) => {
+  if (group) {
+    return (
+      <optgroup key={`optgroup-${label}`} label={label} {...attributes}>
+        {renderOptions(group)}
+      </optgroup>
+    );
+  }
+  return <option key={value} value={value} {...attributes}>{label}</option>;
+});
+
+
+class Input extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.innerRef = props.innerRef; // eslint-disable-line react/prop-types
+    if (process.env.NODE_ENV === 'development' && !this.innerRef) {
```

Comment:
In order to check and see if there is an associated label we must have a ref to the html node. Since we only what warnings in development mode, this component will, create a ref if none is supplied. In production and other modes we shouldn't create a ref that we aren't going to use for anything.

Reply:
What if the user of Input needs access to the input element in the DOM to control focus, etc.? Without the ref in both development and production, it doesn't seem like we'd be able to support that use case?

Reply:
React.forwardRef will allow us to add refs on this component as if it were the raw input node. In a parent component:

`<Input type="text" ref={myInputRef} />`

`myInputRef` will give you reference to the html node.  We won't have any capability to get a ref  to the Input component, but this is okay I think.

Reply:
Yeah, I think we'd only need a ref to the actual html node, not the component itself. forwardRef is still a bit new to me (haven't used it too much), so thanks for that clarification!